data/campaigns/Descent_Into_Darkness/scenarios/06_Return_to_Parthyn.cfg

```diff
@@ -259,53 +279,181 @@
+            message= _ "<i>If I want the people of Parthyn to accept me back, I must be cautious. I must not harm any of them besides Drogan, or they will surely out me for good.</i>"
         [/message]
     [/event]
 
+    # activate side 4 a tiny bit late or they rush the player a bit quickly
+    [event]
+        name=turn 3
+
+        [gold]
+            side=4
+            {QUANTITY amount 40 50 60}
+        [/gold]
+    [/event]
+    [event]
+        name=turn 4
+
+        [gold]
+            side=4
+            {QUANTITY amount 50 70 90}
+        [/gold]
+    [/event]
+
+    # Placing the orc leader
+
+    [event]
+        name=turn 5
+
+        [scroll_to]
+            x,y=29,1
+        [/scroll_to]
+
+        [move_unit_fake]
+            type="Orcish Warlord"
+            side=2
+            x=29,29,28,28,29,30
+            y=1,2,2,3,4,4
+        [/move_unit_fake]
+
+        # Replace the fallback orcish leader with the stored one from the
+        # previous scenario
+        [lua]
+            code = <<
+                local saved_orc_leader = wml.variables.orc_leader_store
+                local fallback_orc_leader = wml.variables.fallback_orc_leader
+
+                if saved_orc_leader ~= nil then
+                    saved_orc_leader.side = 2
+                    saved_orc_leader.facing = "se"
+
+                    local loc = wesnoth.get_starting_location(2)
+                    wesnoth.put_unit(saved_orc_leader, loc[1], loc[2])
+
+                    wml.variables.orc_leader_store = nil
+                else
+                    local loc = wesnoth.get_starting_location(2)
+                    wesnoth.put_unit(fallback_orc_leader, loc[1], loc[2])
+                end
+            >>
+        [/lua]
+
+        [terrain]
+            x,y=30,4
+            terrain=Ke
+        [/terrain]
+        [terrain]
+            x=30,30,29,29
+            y=3,5,4,5
+            terrain=Ce
+        [/terrain]
+        [redraw][/redraw]
+
+        [if]
+            {DID_RTP_HAS_DROGAN}
+            [then]
+                [message]
+                    side=2
+                    canrecruit=yes
+                    message= _ "Grunts! Prepare the camp! We attack the city tonight— wait. You! Necromancer, you followed us all the way here?!"
+                [/message]
+
+                [message]
+                    speaker=Malin Keshar
+                    message= _ "Did you really think you could outrun the dead, who need no rest nor sleep? Prepare to meet your makers, orc. Drogan, I will show you what my new powers are capable of. You will see how foolish you are to oppose me."
+                [/message]
+            [/then]
+            [else]
+                [message]
+                    side=2
+                    canrecruit=yes
+                    message= _ "Grunts! Prepare the camp! We attack the city tonight—wait. You! Necromancer, you followed us all the way here?!"
```

Comment:
This message is duplicated in both parts of the if statement. However, there's a single-space change between the two copies (which I assume isn't intentional).

Reply:
Thanks for pointing that out, I missed it.